tests/spki_tests.rs

```diff
@@ -0,0 +1,615 @@
+                         "rsa-pkcs1-sha256-spki-non-null-params.pem",
+                         &spki::RSA_PKCS1_2048_8192_SHA256,
+                         Err(VerifyWithSPKIError::UnsupportedSignatureAlgorithmForPublicKey));
+test_verify_signature_pem!(
+    test_rsa_pkcs1_sha256_using_id_ea_rsa,
+    "rsa-pkcs1-sha256-using-id-ea-rsa.pem",
+    &spki::RSA_PKCS1_2048_8192_SHA256,
+    Err(VerifyWithSPKIError::UnsupportedSignatureAlgorithmForPublicKey));
+
+/// Our PSS tests that should work.
+test_verify_signature_pem!(
+    test_rsa_pss_sha256_salt32,
+    "ours/rsa-pss-sha256-salt32.pem",
+    &spki::RSA_PSS_2048_8192_SHA256_LEGACY_KEY,
+    Ok(()));
+test_verify_signature_pem!(
+    test_rsa_pss_sha384_salt48,
+    "ours/rsa-pss-sha384-salt48.pem",
+    &spki::RSA_PSS_2048_8192_SHA384_LEGACY_KEY,
+    Ok(()));
+test_verify_signature_pem!(
+    test_rsa_pss_sha512_salt64,
+    "ours/rsa-pss-sha512-salt64.pem",
+    &spki::RSA_PSS_2048_8192_SHA512_LEGACY_KEY,
+    Ok(()));
+test_verify_signature_pem!(
+    test_rsa_pss_sha256_salt32_corrupted_data,
+    "ours/rsa-pss-sha256-salt32-corrupted-data.pem",
+    &spki::RSA_PSS_2048_8192_SHA256_LEGACY_KEY,
+    Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
+test_verify_signature_pem!(
+    test_rsa_pss_sha384_salt48_corrupted_data,
+    "ours/rsa-pss-sha384-salt48-corrupted-data.pem",
+    &spki::RSA_PSS_2048_8192_SHA384_LEGACY_KEY,
+    Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
+test_verify_signature_pem!(
+    test_rsa_pss_sha512_salt64_corrupted_data,
+    "ours/rsa-pss-sha512-salt64-corrupted-data.pem",
+    &spki::RSA_PSS_2048_8192_SHA512_LEGACY_KEY,
+    Err(VerifyWithSPKIError::InvalidSignatureForPublicKey));
+
+test_verify_signature_pem!(
+    test_rsa_using_ec_key, "rsa-using-ec-key.pem",
+    &spki::RSA_PKCS1_2048_8192_SHA256,
+    Err(VerifyWithSPKIError::UnsupportedSignatureAlgorithmForPublicKey));
+test_verify_signature_pem!(test_rsa2048_pkcs1_sha512,
+                       "rsa2048-pkcs1-sha512.pem",
+                       &spki::RSA_PKCS1_2048_8192_SHA512,
+                       Ok(()));
+
+struct TestSignedData {
+    spki: std::vec::Vec<u8>,
+    data: std::vec::Vec<u8>,
+    signature: std::vec::Vec<u8>
+}
+
+fn parse_test_signed_data(file_name: &str) -> TestSignedData {
+    let path =
+    std::path::PathBuf::from(
+        "third-party/chromium/data/verify_signed_data").join(file_name);
+    let file = std::fs::File::open(path).unwrap();
+    let mut lines = std::io::BufReader::new(&file).lines();
+
+    let spki = read_pem_section(&mut lines, "PUBLIC KEY");
+    let data = read_pem_section(&mut lines, "DATA");
+    let signature = read_pem_section(&mut lines, "SIGNATURE");
+
+    TestSignedData {
+        spki: spki,
+        data: data,
+        signature: signature
+    }
+}
+
+type FileLines<'a> = std::io::Lines<std::io::BufReader<&'a std::fs::File>>;
+
+fn read_pem_section(lines: & mut FileLines, section_name: &str)
```

Comment:
I think I understand that we need to copy/paste because these integration tests need to be run against the production version of *ring*, and we don't want these common functions to be part of the public *ring* API. Let's add a comment explaining this to this file.

Another way to do this, which I prefer, would be to put the common logic in a separate source file, e.g. tests/der_tests_common.rs, and then `include!` that file in `ring::der::tests` and in this file. WDYT?

Otherwise, we at least need to add some comments "Keep this in sync with the copy in <other file's relative path>" to both src/der.rs and this file, for each item that is duplicated.